### PR TITLE
efficientnet wrapper gets pretrained model from efficientnet_python package

### PR DESCRIPTION
**Pull request by Akssi**

### CODE REVIEW

**Comment by Olivier-tl** [comment]
We can't use a pretrained model because it breaks the imbalanced dataset assumption as mentioned in the GAMO paper! I think the model should be a torch.nn.module so that we can directly call forward or save it as we would with any other model.

File: models/model_factory.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-from .classification import EfficientNet
+from .classification import EfficientNetWrapper
 from .gans import WGAN
 
 class ModelFactory:
     def create(model_name: str) -> torch.nn.Module:
         if model_name == 'efficientnet':
-            model = EfficientNet()
+            model = EfficientNetWrapper('efficientnet-b7')
```
Comment (Olivier-tl) [inline models/model_factory.py:9]: Actually I think if we use 'efficientnet-l2' the weights aren't pre-trained

File: models/classification/efficient_net_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from efficientnet_pytorch import EfficientNet
+import utils.logging
+
+"""Wrapper on top of efficientnet model implementation
+"""
+class EfficientNetWrapper:
+    def __init__(self, model_name:str):
+        self.model = EfficientNet.from_pretrained(model_name)
```
Comment (Olivier-tl) [inline models/classification/efficient_net_wrapper.py:8]: It could be a "create" method and directly return the instance of the model

Comment (Olivier-tl) [reply models/classification/efficient_net_wrapper.py:8]: or at this point the factory could create it directly haha...

File: models/model_factory.py (hunk 1 above)
Comment (Akssi) [reply models/model_factory.py]: We can use from_name to get untrained version

File: models/classification/efficient_net_wrapper.py (hunk 2 above)
Comment (Akssi) [reply models/classification/efficient_net_wrapper.py]: Indeed factory shoulkd just return it no need for the wrapper

**Comment by Akssi** [comment]
Updated with untrained model + no wrapper

**Review by Olivier-tl — APPROVED** [review]
👌🏻

**Review by wwwnev — APPROVED** [review]

**Pull request merged**